errors/pluginerrors.go

```diff
@@ -0,0 +1,159 @@
+		Description: `Generic error returned when the referrer store fails to perform an operation.`,
+	})
+
+	// ErrorCodeCreateRepositoryFailure is returned when Referrer Store fails to
+	// create a repository object.
+	ErrorCodeCreateRepositoryFailure = Register("errcode", ErrorDescriptor{
+		Value:       "CREATE_REPOSITORY_FAILURE",
+		Message:     "create repository failure",
+		Description: "Failed to create repository. Check the repository config for more details.",
+	})
+
+	// ErrorCodeRepositoryOperationFailure is returned when a repository
+	// operation fails.
+	ErrorCodeRepositoryOperationFailure = Register("errcode", ErrorDescriptor{
+		Value:       "REPOSITORY_OPERATION_FAILURE",
+		Message:     "repository operation failure",
+		Description: `The repository operation fails. Check the repository for more details.`,
+	})
+
+	// ErrorCodeManifestInvalid is returned if fetched manifest is invalid.
+	ErrorCodeManifestInvalid = Register("errcode", ErrorDescriptor{
+		Value:       "MANIFEST_INVALID",
+		Message:     "manifest invalid",
+		Description: `The manifest is invalid. Check the manifest for more details.`,
+	})
+
+	// ErrorCodeReferrersNotFound is returned if there is no ReferrerStore set.
+	ErrorCodeReferrersNotFound = Register("errcode", ErrorDescriptor{
+		Value:       "REFERRERS_NOT_FOUND",
+		Message:     "referrers not found",
+		Description: "No referrers are found. Check the artifact is configured correctly.",
+	})
+
+	// Generic errors happen in plugins
+
+	// ErrorCodePluginInitFailure is returned when it fails to initialize a plugin.
+	ErrorCodePluginInitFailure = Register("errcode", ErrorDescriptor{
+		Value:       "PLUGIN_INIT_FAILURE",
+		Message:     "plugin init failure",
+		Description: "The plugin fails to initialize. Check the plugin config for more details.",
+	})
+
+	// ErrorCodePluginNotFound is returned when it cannot find the required plugin.
+	ErrorCodePluginNotFound = Register("errcode", ErrorDescriptor{
+		Value:       "PLUGIN_NOT_FOUND",
+		Message:     "plugin not found",
+		Description: "No plugin was found. Check the plugin config for more details.",
+	})
+
+	// ErrorCodeDownloadPluginFailure is returned when it fails to download a
+	// plugin.
+	ErrorCodeDownloadPluginFailure = Register("errcode", ErrorDescriptor{
+		Value:       "DOWNLOAD_PLUGIN_FAILURE",
+		Message:     "download plugin failure",
+		Description: "Failed to download plugin. Check if the provided source is correct.",
+	})
+
+	// ErrorCodeCertInvalid is returned when provided certificates are invalid.
+	ErrorCodeCertInvalid = Register("errcode", ErrorDescriptor{
+		Value:       "CERT_INVALID",
+		Message:     "cert invalid",
+		Description: "The certificate is invalid. Check the certificate for more details.",
+	})
+
+	// ErrorCodeProviderNotFound is returned when a required provider cannot be
+	// found.
+	ErrorCodeProviderNotFound = Register("errcode", ErrorDescriptor{
+		Value:       "PROVIDER_NOT_FOUND",
```

Comment:
suggestion to improve this message, " which provider " is it the certificate provider?